tests/rustdoc/sidebar-link-sort.rs

```diff
@@ -0,0 +1,13 @@
+// issue #56018: "Implementations on Foreign Types" sidebar items should be sorted by name
+#![crate_name = "foo"]
+#![no_std]
+
+pub trait Foo {}
+
+//@ has 'foo/trait.Foo.html'
+//@ snapshot sidebar - '//*[@class="sidebar-elems"]'
```

Comment:
That sadly doesn't enforce really well the sorting order here since a `--bless` will just update the html file.

Reply:
I didn't get you point, we treat the HTML files as compared files in other unit testing, we can see the names and links in HTML file (`tests/rustdoc/sidebar-link-sort.sidebar.html`) are sorted as assumed, if anything broke the sorting order, the HTML files will be changed.

Reply:
I mentioned that it should likely be a gui test (`tests/rustdoc-gui`) [here](https://github.com/rust-lang/rust/pull/137106#discussion_r1957279312).

Reply:
added test case in `tests/rustdoc-gui`.